streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -35,48 +36,89 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Optional;
-import java.util.Set;
+import java.util.stream.Collectors;
 
 import static java.lang.String.format;
-import static java.util.Collections.emptyMap;
-import static java.util.Collections.unmodifiableList;
 import static org.apache.kafka.streams.processor.internals.StateManagerUtil.CHECKPOINT_FILE_NAME;
 import static org.apache.kafka.streams.processor.internals.StateManagerUtil.converterForStore;
 import static org.apache.kafka.streams.processor.internals.StateRestoreCallbackAdapter.adapt;
 
-
+/**
+ * ProcessorStateManager is the source of truth for the current offset for each state store,
+ * which is either the read offset during restoring, or the written offset during normal processing.
+ *
+ * The offset is initialized as null when the state store is registered, and then it can be updated by
+ * loading checkpoint file, restore state stores, or passing from the record collector's written offsets.
+ *
+ * When checkpointing, if the offset is not null it would be written to the file.
+ *
+ * The manager is also responsible for restoring state stores via their registered restore callback,
+ * which is used for both updating standby tasks as well as restoring active tasks.
+ */
 public class ProcessorStateManager implements StateManager {
```

Comment:
Re: `Q: I am wondering, if it would be cleaner`: we do not register all state stores within a state manager to the changelog reader, but only those log-enabled ones with a given changelog. We can, still, register a state manager into a change logger and let the state manager to return a subset of stores that are log-enabled, I just felt it is a bit larger scope and did not do this refactoring here.

Yes I can, how about doing the renaming in a separate PR? Github sometimes confuses with file renaming and translate it to a deletefile + createNewFile which could make reviewing much harder. I will do a renaming with no class changes at all just to ease reviews.

Reply:
I think reducing dependencies is always a good thing to do, because it simplifies refactoring and testing. I like the idea that the state manager returns only the log-enabled state stores. See my comment below regarding using two maps, one for log-enabled and one for log-disabled state stores.